src/Kanvas/Filesystem/Workflows/Activities/DownloadImageActivity.php

```diff
@@ -0,0 +1,87 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Kanvas\Filesystem\Workflows\Activities;
+
+use Baka\Contracts\AppInterface;
+use Illuminate\Database\Eloquent\Model;
+use Illuminate\Support\Facades\Storage;
+use Illuminate\Support\Str;
+use Kanvas\Filesystem\Services\FilesystemServices;
+use Kanvas\Workflow\Contracts\WorkflowActivityInterface;
+use Workflow\Activity;
+
+class DownloadImageActivity extends Activity implements WorkflowActivityInterface
+{
+    public function execute(Model $entity, AppInterface $app, array $params): array
+    {
+        $file = file_get_contents($entity->url);
```

Comment:
@FredPeal move the image actvity logic to a action and call the action withing the activity so you can use the same method in multiple location if needed